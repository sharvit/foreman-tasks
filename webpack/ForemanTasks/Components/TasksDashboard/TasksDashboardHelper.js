import URI from 'urijs';
import { getURIQuery } from 'foremanReact/common/helpers';

import { TASKS_DASHBOARD_AVAILABLE_TIMES_TEXT } from './TasksDashboardConstants';

export const getTimeText = time => TASKS_DASHBOARD_AVAILABLE_TIMES_TEXT[time];

export const getQueryFromUrl = () => {
  const query = {};
  const uriQuery = getURIQuery(window.location.href);

  ['state', 'result', 'mode', 'time'].forEach(f => {
    if (uriQuery[f]) query[f] = uriQuery[f];
  });

  return query;
};

export const resolveQuery = query => {
  const uri = new URI(window.location.href);
  const { search } = uri.query(true);

  const data = { search, ...query, page: 1 };
  uri.query(URI.buildQuery(data, true));
  window.Turbolinks.visit(uri.toString());
};
