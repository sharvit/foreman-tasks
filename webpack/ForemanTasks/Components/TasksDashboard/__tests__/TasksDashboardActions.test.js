import { testActionSnapshotWithFixtures } from 'react-redux-test-utils';
import API from 'foremanReact/API';
import { timeToHoursNumber } from '../TasksDashboardHelper';
import {
  initializeDashboard,
  updateTime,
  updateQuery,
  fetchTasksSummary,
} from '../TasksDashboardActions';

jest.mock('foremanReact/API');
jest.mock('../TasksDashboardHelper');

const correctTime = 'H24';
const wrongTime = 'H25';

timeToHoursNumber.mockImplementation(arg => arg);
API.get.mockImplementation(async path => {
  if (path === `/foreman_tasks/tasks/summary/${correctTime}`) {
    return { data: 'some-data' };
  }

  throw new Error('wrong time');
});

const fixtures = {
  'should initialize-dashboard': () =>
    initializeDashboard({ time: 'some-time', query: 'some-query' }),
  'should update-time': () => updateTime('some-time'),
  'should update-query': () => updateQuery('some-query'),
  'should fetch-tasks-summary and success': () =>
    fetchTasksSummary(correctTime),
  'should fetch-tasks-summary and fail': () => fetchTasksSummary(wrongTime),
};

describe('TasksDashboard - Actions', () =>
  testActionSnapshotWithFixtures(fixtures));
