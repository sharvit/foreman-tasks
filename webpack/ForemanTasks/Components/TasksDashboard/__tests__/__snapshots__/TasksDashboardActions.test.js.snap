// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksDashboard - Actions should fetch-tasks-summary and fail 1`] = `
Array [
  Array [
    Object {
      "type": "FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_REQUEST",
    },
  ],
  Array [
    Object {
      "payload": [Error: wrong time],
      "type": "FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_FAILURE",
    },
  ],
]
`;

exports[`TasksDashboard - Actions should fetch-tasks-summary and success 1`] = `
Array [
  Array [
    Object {
      "type": "FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_REQUEST",
    },
  ],
  Array [
    Object {
      "payload": "some-data",
      "type": "FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_SUCCESS",
    },
  ],
]
`;

exports[`TasksDashboard - Actions should initialize-dashboard 1`] = `
Object {
  "payload": Object {
    "query": "some-query",
    "time": "some-time",
  },
  "type": "FOREMAN_TASKS_DASHBOARD_INIT",
}
`;

exports[`TasksDashboard - Actions should update-query 1`] = `
Array [
  Array [
    Object {
      "payload": "some-query",
      "type": "FOREMAN_TASKS_DASHBOARD_UPDATE_QUERY",
    },
  ],
]
`;

exports[`TasksDashboard - Actions should update-time 1`] = `
Object {
  "payload": "some-time",
  "type": "FOREMAN_TASKS_DASHBOARD_UPDATE_TIME",
}
`;
